src/Tasks/CultureInfoCache.cs

```diff
@@ -19,35 +19,36 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     internal static class CultureInfoCache
     {
-        private static readonly HashSet<string> ValidCultureNames;
+        private static readonly HashSet<string> ValidCultureNames = InitializeValidCultureNames();
 
-        static CultureInfoCache()
+        static HashSet<string> InitializeValidCultureNames()
         {
-            ValidCultureNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-
+            HashSet<string> validCultureNames = new(StringComparer.OrdinalIgnoreCase);
 #if !FEATURE_CULTUREINFO_GETCULTURES
             if (!AssemblyUtilities.CultureInfoHasGetCultures())
             {
-                ValidCultureNames = HardcodedCultureNames;
-                return;
+                validCultureNames = HardcodedCultureNames;
+                return validCultureNames;
```

Comment:
nit:
```suggestion
                return HardcodedCultureNames;
```

And move this to the top of the function?